iml-gui/crate/src/components/command_modal.rs

```diff
@@ -193,6 +200,23 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>)
                 schedule_fetch_tree(model, orders);
             }
         }
+        Msg::CancelJob(job_id) => {
+            if let Some(job) = model.jobs.get(&job_id) {
+                if let Some(ct) = find_cancel_transition(job) {
```

Comment:
`ct` is not used? Maybe `if find_cancel_transition(job).is_some()` then?

Reply:
Might be `model.jobs.get(&job_id).and_then(|j| find_cancel_transition(j)).is_some()`

Reply:
It is used two lines down: `TransitionState(ct.state.clone())`

Reply:
IMHO, still could be less nested as

```rust
if let Some(ct) = model.jobs.get(&job_id).and_then(|j| find_cancel_transition(j)) {
..
}